Xamarin.PropertyEditing.Mac/Controls/BindingEditor/BindingPathSelectorControl.cs

```diff
@@ -0,0 +1,339 @@
+				switch (target) {
+					case PropertyTreeElement propertyTreeElement:
+						IReadOnlyCollection<PropertyTreeElement> propertyTrees = propertyTreeElement.Children.Task.Result;
+						childCount = propertyTrees.Count;
+						break;
+
+					case string targetName:
+						childCount = this.viewModel.Count ();
+						break;
+
+					default:
+						childCount = 0;
+						break;
+				}
+			}
+
+			return childCount;
+		}
+
+		public override NSObject GetChild (NSOutlineView outlineView, nint childIndex, NSObject item)
+		{
+			if (childIndex == 0 && item == null) {
+				return new NSObjectFacade (targetName);
+			}
+
+			if (item is NSObjectFacade objectFacade) {
+				var target = objectFacade.Target;
+				switch (target) {
+					case PropertyTreeElement propertyTreeElement:
+						IReadOnlyCollection<PropertyTreeElement> propertyTrees = propertyTreeElement.Children.Task.Result;
+						return new NSObjectFacade (propertyTrees.ElementAt ((int)childIndex));
+
+					case string targetName:
+						return new NSObjectFacade (this.viewModel.ElementAt ((int)childIndex));
+
+					default:
+						return null;
+				}
+
+			}
+			return null;
+		}
+
+		public override bool ItemExpandable (NSOutlineView outlineView, NSObject item)
+		{
+			if (item is NSObjectFacade objectFacade) {
+				var target = objectFacade.Target;
+				switch (target) {
+					case PropertyTreeElement propertyTreeElement:
+						IReadOnlyCollection<PropertyTreeElement> propertyTrees = propertyTreeElement.Children.Task.Result;
+						return propertyTrees.Count > 0;
+
+					case string targetName:
+						return true;
+
+					default:
+						return false;
+				}
+			}
+
+			return false;
+		}
+	}
+
+	internal class BindingPathSelectorControl : NSView
+	{
+		private PathOutlineView pathOutlineView;
+		internal const string PathSelectorColumnColId = "PathSelectorColumn";
+		private const float spacing = 10f;
+
+        public NSTextField CustomPath { get; }
+
+        public BindingPathSelectorControl (CreateBindingViewModel viewModel)
+		{
+			TranslatesAutoresizingMaskIntoConstraints = false;
+
+			var customCheckBox = new NSButton {
+				ControlSize = NSControlSize.Small,
+				Font = NSFont.FromFontName (PropertyEditorControl.DefaultFontName, PropertyEditorControl.DefaultFontSize),
+				Title = Properties.Resources.Custom,
+				TranslatesAutoresizingMaskIntoConstraints = false,
+			};
+
+			customCheckBox.SetButtonType (NSButtonType.Switch);
+
+			AddSubview (customCheckBox);
+			AddConstraints (new[] {
+				NSLayoutConstraint.Create (customCheckBox, NSLayoutAttribute.Top, NSLayoutRelation.Equal, this, NSLayoutAttribute.Top, 1f, 8f),
+				NSLayoutConstraint.Create (customCheckBox, NSLayoutAttribute.Left, NSLayoutRelation.Equal, this, NSLayoutAttribute.Right, 1f, -90f),
+			});
+
+			this.CustomPath = new NSTextField {
+				ControlSize = NSControlSize.Mini,
+				Enabled = false,
+				Font = NSFont.FromFontName (PropertyEditorControl.DefaultFontName, PropertyEditorControl.DefaultFontSize),
```

Comment:
Are we sure `Mini` is the right size here? The text sits at the top of what looks like a full or at least small size control in the actual window (as it's using `DefaultControlHeight` which matches what it'd be for `Small`: 18.) These properties are also effectively deprecated, if it's `Mini` or `Small`, use the system method for getting the correct size (which `DefaultFontSize` isn't correct for `Mini`).

```suggestion
				Font = NSFont.SystemFontOfSize (NSFont.SystemFontSizeForControlSize (NSControlSize.Mini)),
```

Reply:
This should have been small.